core/types.go

```diff
@@ -260,14 +260,38 @@ type RenewalInfo struct {
 // using a very simple renewal calculation: calculate a point 2/3rds of the way
 // through the validity period, then give a 2-day window around that. Both the
 // `issued` and `expires` timestamps are expected to be UTC.
-func RenewalInfoSimple(issued time.Time, expires time.Time) *RenewalInfo {
+func RenewalInfoSimple(issued time.Time, expires time.Time, now time.Time) *RenewalInfo {
 	validity := expires.Add(time.Second).Sub(issued)
 	renewalOffset := validity / time.Duration(3)
 	idealRenewal := expires.Add(-renewalOffset)
+	windowStart := idealRenewal.Add(-24 * time.Hour)
+	windowEnd := idealRenewal.Add(24 * time.Hour)
+
+	if expires.Before(now) {
+		// If the cert has expired, return `RenewalInfoImmediate` instead.
+		return RenewalInfoImmediate(now)
+	}
```

Comment:
This is redundant: if we're clamping `windowEnd` to `expires`, then an expired cert will always be given a window that is wholly in the past. No need to special case it here. This is why I suggested the removal of this stanza in my previous review.

If you want to keep it anyway, move it above the computation of the ideal window, since none of that computation is necessary for this stanza.

Reply:
Oh, that makes sense.  I just removed it.